pybossa/view/bloomberg.py

```diff
@@ -70,11 +71,30 @@ def handle_bloomberg_response():
         current_app.logger.error('BSSO auth error(s): %s %s', errors, error_reason)
         flash(gettext('There was a problem during the sign in process.'), 'error')
         return redirect(url_for('home.home'))
-    if auth.is_authenticated:
+    elif auth.is_authenticated:
         attributes = auth.get_attributes()
         user = user_repo.get_by(email_addr=unicode(attributes['emailAddress'][0]).lower())
         return _sign_in_user(user, next_url=request.form.get('RelayState'))
-    else:
+    elif not auth.is_authenticated:
         current_app.logger.error('BSSO authentication failed')
         flash(gettext('Authentication failed.'), 'error')
         return redirect(url_for('home.home'))
+    else:
+        attributes = auth.get_attributes()
+        user_data = {}
+        try:
+            user_data['fullname']   = attributes['FirstName'][0] + " " + attributes['LastName'][0]
+            user_data['email_addr'] = attributes['emailAddress'][0]
+            user_data['name']       = attributes['LoginID'][0]
+            user_data['password']   = u"pAssw0rd"
+            create_account(user_data)
+            user = user_repo.get_by(email_addr=unicode(attributes['emailAddress'][0]).lower())
+            return _sign_in_user(user, next_url=request.form.get('RelayState'))
+        except DBIntegrityError as dberror:
```

Comment:
This exception block appears very similar to the general one below under `except Exception as ex`. The only difference is a slight wording in the message to the user. Perhaps we could consider just using the general, rather than having two separate ones that perform similar functionality. That would simplify the logic under a single exception handler.

Reply:
I was keeping the text distinct intentionally to make debugging easier. For example if a user refers to one text over the other it will be easier to see if it is a DB or other error. Would you like me to make it more general?

Reply:
I will just do as you suggest.

Reply:
Regarding debugging, the type of exception should be [included](https://stackoverflow.com/a/37992988) in the `logger.error` output. Feel free to refer to other similar code in this repository for examples of logging errors.

Also, it's recommended to unit test the code (for example, add an intentional exception) to confirm the output that the log will create is as expected. Thanks!

Reply:
`logger.exception` includes the stack trace and the exception type, so you can see right away what caused the problem.